src/models/sourcecodemodel.cpp

```diff
@@ -34,6 +34,7 @@ void SourceCodeModel::clear()
 void SourceCodeModel::setDisassembly(const DisassemblyOutput& disassemblyOutput)
 {
     m_numLines = 0;
```

Comment:
this was probably wrong before, but we need to `beginResetModel` early and ensure that we always call `endResetModel` then too. I suggest we rewrite this by leveraging `qScopeGuard`:

```
beginResetModel();
auto guard = qScopeGuard([this]() { endResetModel(); });

m_numLines = 0;
m_numTypes = 0;
...
```

Reply:
Done.